omp/preconditioner/sor_kernels.cpp

```diff
@@ -0,0 +1,67 @@
+// SPDX-FileCopyrightText: 2017 - 2024 The Ginkgo authors
```

Comment:
you may need to use `GKO_KERNEL` for the lambda function.
Maybe also for the actual kernel?

Reply:
I feel like this complicates our unified setup again, and it's not using the unified kernels as intended.
We use the unified kernels, when the kernel implementation are identical, which is not the case here. I think it's close, but it would require some adjustments of the factorization kernels. If that's feasible, I would prefer to do it in another PR.